anvil-python/anvil/commands/haproxy.py

```diff
@@ -183,21 +227,30 @@ def extra_tfvars(self) -> dict[str, Any]:
             self.client, self._HAPROXY_CONFIG
         )
 
-        cert_filepath = variables["ssl_cert"]
-        key_filepath = variables["ssl_key"]
-        if cert_filepath != "" and key_filepath != "":
-            with open(cert_filepath) as cert_file:
+        if variables["tls_mode"] != "disabled":
+            variables["haproxy_port"] = 443
+            variables["haproxy_services_yaml"] = self.get_tls_services_yaml(
+                variables["tls_mode"]
+            )
+            if not variables["ssl_cert"] or not variables["ssl_key"]:
+                raise TerraformException(
+                    "Both ssl_cert and ssl_key must be provided when enabling TLS"
+                )
+            with open(variables["ssl_cert"]) as cert_file:
                 variables["ssl_cert_content"] = cert_file.read()
-            with open(key_filepath) as key_file:
+            with open(variables["ssl_key"]) as key_file:
                 variables["ssl_key_content"] = key_file.read()
-            variables["haproxy_port"] = 443
-            variables["haproxy_services_yaml"] = self.get_tls_services_yaml()
+            if variables["ssl_cacert"]:
+                with open(variables["ssl_cacert"]) as cacert_file:
+                    variables["ssl_cacert_content"] = cacert_file.read()
```

Comment:
I am curious how haproxy is going to use cacert. Unless it is not using it at all. Could you please explain why this is not used only by maas-region?

Reply:
This is here for if the user selects `passthrough` as the `tls_mode`. All of the cert/key/cacert get passed on to `maas-region` if this is the case. So, haproxy doesn't ever use the cacert, but since we're asking TLS questions here when HA proxy is enabled, I figured it's best that all of the TLS questions come at the same time, even if the answers are intended to be used by `maas-region`

Reply:
oh, I now see the problem here--I think this logic can be removed from `extra_tfvars` here